app/models/event.rb

```diff
@@ -118,6 +121,48 @@ class Event < ApplicationRecord
       .where("flipper_gates.feature_key = ? AND flipper_gates.key = ?", flag, "actors")
   }
 
+  def ancestor_ids
+    [id] + Event.connection.execute(<<-SQL).map { |row| row["id"] }
```

Comment:
is an event an ancestor of itself?

Reply:
This is how Ruby handles ancestors:

<img width="516" height="87" alt="image" src="https://github.com/user-attachments/assets/6b9ebf2c-1b1f-46b6-ad4c-72248f4cd64c" />

Reply:
Interesting TIL - if we can, I'd vote to avoid replicating this because I think it's confusing or doing what Gary suggested: https://github.com/hackclub/hcb/pull/9739/files#r2226423504